src/clinical_etl/CSVConvert.py

```diff
@@ -568,15 +568,32 @@ def load_manifest(manifest_file):
         sys.exit(f"Manifest file not found at provided path: {manifest_file}")
 
     if "identifier" in manifest:
-        identifier = manifest["identifier"]
-    if "schema" in manifest:
-        schema = manifest["schema"]
+        result["identifier"] = manifest["identifier"]
+    if "schema" not in manifest:
+        sys.exit("Need to specify an OpenAPI schema as 'schema' in the manifest file, "
+                 "see README for more details.")
+    if "schema_class" in manifest:
+        schema_class = manifest["schema_class"]
+
+    # programatically load schema class based on manifest value:
+    # schema class definition will be in a file named schema_class.lower()
+    schema_mod = importlib.import_module(f"clinical_etl.{schema_class.lower()}")
+    schema = getattr(schema_mod, schema_class)(manifest["schema"])
+    if schema.json_schema is None:
+        sys.exit(f"Could not read an openapi schema at {manifest['schema']};\n"
+              f"please check the url in the manifest file links to a valid openAPI schema.")
+    result["schema"] = schema
+
     if "mapping" in manifest:
         mapping_file = manifest["mapping"]
         manifest_dir = os.path.dirname(os.path.abspath(manifest_file))
         mapping_path = os.path.join(manifest_dir, mapping_file)
         if os.path.isabs(mapping_file):
             mapping_path = manifest_file
+        result["mapping"] = mapping_path
+
+    if "reference_date" in manifest:
```

Comment:
I think it would be good to add some kind of error check, nice message and exit like for other detectable errors in the manifest

Reply:
Thinking about this more , the date interval is specific to MoH so maybe generic error and exit doesn't make sense.

Reply:
unless we did something like if schema_class == MoHSchema and no reference date? Not really sure of best approach but don't like not catching the error and letting the user know how they can fix it

Reply:
I don't know the best way to do this either...

Reply:
yeah I guess we leave it for now and we can think about it in the future